openedx_learning/apps/authoring/collections/api.py

```diff
@@ -204,3 +205,33 @@ def get_collections(learning_package_id: int, enabled: bool | None = True) -> Qu
     if enabled is not None:
         qs = qs.filter(enabled=enabled)
     return qs.select_related("learning_package").order_by('pk')
+
+
+def remove_unpublished_from_collections(learning_package_id: int) -> None:
+    """
+    Removes all unpublished entities from collections for a
+    given learning package.
```

Comment:
@ChrisChV Can you please add an explanation/example of _why_ we need this API, and what it's used for?

Reply:
Also as part of the PR and commit message please :)

Reply:
- Comment added here: https://github.com/openedx/openedx-learning/pull/252/commits/3b940ce9d9ad482fe83b5613650c36655c9c8694
- Commit message added here: https://github.com/openedx/openedx-learning/pull/252/commits/cafe2a3f9a8c98c44fe3472bf3d7ecaf3a4a2cfa